lib/flop.ex

```diff
@@ -772,11 +772,33 @@ defmodule Flop do
     if count = opts[:count] do
       count
     else
-      q = opts[:count_query] || q
+      q = opts[:count_query] || count_query(q)
       apply_on_repo(:aggregate, "count", [filter(q, flop, opts), :count], opts)
     end
   end
 
+  defp count_query(query) do
+    query =
+      query
+      |> Query.exclude(:preload)
+      |> Query.exclude(:order_by)
+      |> Query.exclude(:select)
+
+    case query do
+      %{distinct: %{expr: expr}} = query
+      when expr == true or (is_list(expr) and expr != []) ->
+        Query.subquery(query)
+
+      %{group_bys: group_bys} = query when group_bys != [] ->
+        query
+        |> Query.select(%{dont_care: true})
```

Comment:
Could this just be `Query.select(1)`?

Reply:
IIRC this would be an error [here](https://github.com/elixir-ecto/ecto/blob/29e9566fa36cdc9a8ed9151d1db377cff26a1331/lib/ecto/query/planner.ex#L369) because of some Ecto validations (I think it's a limitation of Ecto, not SQL) and that's why Scrivener had some logic that tries to select the first field from a `group_by`)

Reply:
Looks like it can be `Query.select(%{})`, I'll change to that